plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -44,6 +44,32 @@ public ImportApplicationWizard() {
 	}
 
 	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		Connection connection = UIUtils.getFirstElement(selection, Connection.class);
+		if (connection != null) {
+			model.setConnection(connection);
+		} else {
+			IResource resource = UIUtils.getFirstElement(selection, IResource.class);
```

Comment:
i agree on this but the client API has a gotcha that prevents uniformous treatement of resources and projects:
* on a IResource that is NOT a project you get the project via .getProject()
* on a IProject which IS a project you get null from .getProject(), you need to take the resource itself